File: minigit-4/ts/src/commands/rev-parse.ts
// rev-parse command - Resolve revisions

import { findRepoRoot } from '../utils';
import { resolveRevision } from '../refs';

export function revParse(args: string[]): number {
  const repoRoot = findRepoRoot();
  if (!repoRoot) {
    console.error('fatal: not a minigit repository');
    return 1;
  }

  if (args.length === 0) {
    console.error('fatal: revision required');
    return 1;
  }

  const rev = args[0];
  const sha = resolveRevision(repoRoot, rev);

  if (!sha) {
    console.error(`fatal: ambiguous argument '${rev}': unknown revision`);
    return 1;
  }

  console.log(sha);
  return 0;
}

File: minigit-4/py/commands/rev_parse.py
"""rev-parse command - Resolve revision to SHA."""

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent))

from utils import find_repo_root
from objects import resolve_object


def run(args):
    """Resolve revision to SHA."""
    repo_root = find_repo_root()
    if repo_root is None:
        print("fatal: not a minigit repository", file=sys.stderr)
        return 1

    if not args:
        print("fatal: no revision specified", file=sys.stderr)
        return 1

    for arg in args:
        if arg.startswith('-'):
            # Skip flags for now
            continue

        try:
            sha = resolve_object(arg, repo_root)
            print(sha)
        except ValueError as e:
            print(f"fatal: ambiguous argument '{arg}': unknown revision", file=sys.stderr)
            return 1

    return 0
